Regression: Gradle plugin requires GRAALVM_HOME to be set

I think this is a regression we previously saw and fix with @melix but it seems to be back with 0.9.2. When using https://sdkman.io to install and use GraalVM, `GRAALVM_HOME` env variable is not set and when using the Gradle plugin and I run `./gradlew clean nativeBuild` I get the following error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeBuild'.
> Expected to find native-image executable but it wasn't found. Make sure to declare the GRAALVM_HOME environment variable or install GraalVM with native-image in a standard location recognized by Gradle Java toolchain support
```

The same use case works correctly with 0.9.1. If I set `GRAALVM_HOME` with 0.9.2 it works.

I **can** reproduce the issue with
- plugin version `0.9.2`
- JDK 8 on the commandline (either `sdk use java 21.2.0.r8-grl` or `sdk use java 8.0.292.hs-adpt`)
- Gradle 6.8.3

I **can't** reproduce the issue with
- plugin version `0.9.2`
- JDK 8 on the commandline (either `sdk use java 21.2.0.r8-grl` or `sdk use java 8.0.292.hs-adpt`)
- Gradle 7.1.1

I have used `gradle wrapper --gradle-version 6.8.3` to generate the Gradle wrapper files and altered the line `id 'org.graalvm.buildtools.native'` to also contain `version '0.9.2'`.


Another observation with Gradle 7.1.1:
With plugin version `0.9.2` and a non-GaalVM (`sdk use java 8.0.292.hs-adpt`) on the command line it even works fine without defining a toolchain that requires GraalVM. How is that?

With plugin version `0.9.1` and a non-GaalVM (`sdk use java 8.0.292.hs-adpt`) on the command line it reports the missing `native-image` which is more expected:
```
> Task :nativeBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeBuild'.
> A problem occurred starting process 'command '~/.sdkman/candidates/java/8.0.292.hs-adpt/bin/native-image''
```

--- a/native-gradle-plugin/src/functionalTest/groovy/org/graalvm/buildtools/gradle/NativeImageOptionsTest.groovy
+++ b/native-gradle-plugin/src/functionalTest/groovy/org/graalvm/buildtools/gradle/NativeImageOptionsTest.groovy
@@ -1,0 +1,44 @@
+package org.graalvm.buildtools.gradle
+
+
+import org.gradle.testkit.runner.GradleRunner
+import spock.lang.Issue
+import spock.lang.Specification
+import spock.lang.TempDir
+
+import java.nio.file.Path
+
+class NativeImageOptionsTest extends Specification {
+    @TempDir
+    Path testDirectory
+
+    @Issue("https://github.com/graalvm/native-build-tools/issues/109")
+    def "toolchain defaults to the current Java version"() {
+        when:
+        def runner = GradleRunner.create()
+                .forwardStdOutput(new PrintWriter(System.out))
+                .forwardStdError(new PrintWriter(System.err))
+                .withPluginClasspath()
+                .withProjectDir(testDirectory.toFile())
+
+
+        def buildFile = testDirectory.resolve("build.gradle")
+        buildFile.text = """
+            plugins {
+                id 'java'
+                id 'org.graalvm.buildtools.native'
+            }
+            
+            assert nativeBuild.javaLauncher
+                .get()
+                .metadata
+                .languageVersion
+                .toString() == JavaVersion.current().majorVersion
+        """
+
+        runner.build()
+
+        then:
+        noExceptionThrown()
+    }
+}

--- a/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java
+++ b/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java
@@ -43,6 +43,7 @@ package org.graalvm.buildtools.gradle.dsl;
 
 import org.graalvm.buildtools.gradle.internal.GradleUtils;
 import org.gradle.api.Action;
+import org.gradle.api.JavaVersion;
 import org.gradle.api.Project;
 import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.api.model.ObjectFactory;
@@ -210,7 +211,7 @@ public abstract class NativeImageOptions {
         getImageName().convention(defaultImageName);
         getJavaLauncher().convention(
                 toolchains.launcherFor(spec -> {
-                    spec.getLanguageVersion().set(JavaLanguageVersion.of(11));
+                    spec.getLanguageVersion().set(JavaLanguageVersion.of(JavaVersion.current().getMajorVersion()));
                     if (GradleUtils.isAtLeastGradle7()) {
                         spec.getVendor().set(JvmVendorSpec.matching("GraalVM"));
                     }

